edge-util/src/Microsoft.Azure.Devices.Edge.Util/Proxy.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System;
+    using System.Net;
+    using Microsoft.Extensions.Logging;
+
+    public static class Proxy
+    {
+        public static Option<IWebProxy> Parse(string proxyUri, ILogger logger)
+        {
+            if (string.IsNullOrWhiteSpace(proxyUri))
+            {
+                return Option.None<IWebProxy>();
+            }
+
+            var uri = new Uri(proxyUri);
```

Comment:
Nit: Can you please add a comment explaining the 3 types of proxy URIs...

Reply:
I added a sample representative URI as a comment for each of the three types, I hoped that was clear enough. You want more?

Reply:
yup, saw it later.. this is good enough. thx